### ISSUE TITLE
Request: Friend Invites & Rich Presence

### ISSUE THREAD

**Issue by noblemaster** [code-disaster/steamworks4j#38]
This is a great library! Thanks :-D

Anyhow, I got a request. In order to invite friends to games, the following methods would really come in handy. With rich presence (2) you can inform others about a game you are playing & people can connect to it. The invite functionality (1) allows to directly invite players to a game! This obviously only makes sense for multiplayer games.

Implementation Required:

    (1) SteamFriends.inviteUserToGame 
    (2) SteamFriends.setRichPresence
    (2) SteamFriendsCallback.onGameRichPresenceJoinRequested

Thanks for the consideration :)

**Comment by nmeyvis** [comment]
`SteamMatchmaking.inviteUserToLobby`

Might help with (1)

**Comment by noblemaster** [comment]
I am aware of that function, but that's not the same.

**Comment by code-disaster** [comment]
Methods added, see 7000a9b.

**Comment by noblemaster** [comment]
Looks great :-D  Do you have SNAPSHOT builds for testing? I can run some tests! :)

**Comment by code-disaster** [comment]
I did not release any snapshot builds so far, but you can easily build your own. Just clone the repository and run "mvn install" in the root folder. This will build 1.6.1-SNAPSHOT and install it to your local Maven cache.

**Comment by noblemaster** [comment]
I gave it a try! Works nicely - Thanks :-D

Anyhow, I found a problem. If you pass in a 'null'-value to `setRichPresence(key, <null>)`, the JVM crashes as follows:

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  EXCEPTION_ACCESS_VIOLATION (0xc0000005) at pc=0x6dac1e19, pid=20108, tid=3112
#
# JRE version: Java(TM) SE Runtime Environment (7.0_60-b19) (build 1.7.0_60-b19)
# Java VM: Java HotSpot(TM) Client VM (24.60-b09 mixed mode windows-x86 )
# Problematic frame:
# V  [jvm.dll+0xc1e19]
#
# Failed to write core dump. Minidumps are not enabled by default on client versions of Windows
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.sun.com/bugreport/crash.jsp
#

...

Stack: [0x0a440000,0x0a490000],  sp=0x0a48f664,  free space=317k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
V  [jvm.dll+0xc1e19]
C  [steamworks4j.dll+0x804b]
j  com.codedisaster.steamworks.SteamFriends.setRichPresence(JLjava/lang/String;Ljava/lang/String;)Z+0
j  com.codedisaster.steamworks.SteamFriends.setRichPresence(Ljava/lang/String;Ljava/lang/String;)Z+6

```

Based on the official documentation, passing in <null> or an empty string removes the key. Anyhow, I am using clearRichPresence() now which works just fine. Haven't tested it but assume an empty string might do the trick as well!?

**Comment by code-disaster** [comment]
Thanks for testing, and nice find. Yes, passing "" should work, but I'll have a look if there is a convenient way to allow null values.

**Comment by noblemaster** [comment]
I guess you could just check if the value that is passed in was 'null' and if yes, convert it to ""! For Steam it's one & the same anyway.

**Comment by code-disaster** [comment]
I released 1.6.1, including this 'null' value workaround.

**Comment by noblemaster** [comment]
Great & Thanks! That was very quick :-D
